tests/unittests/sources/test_smartos.py

```diff
@@ -412,188 +407,186 @@ def close_transport(self):
         self._is_open = False
 
 
-class TestSmartOSDataSource(FilesystemMockingTestCase):
-    jmc_cfact = None
-    get_smartos_environ = None
+@pytest.fixture
+def legacy_user_d(tmp_path):
+    legacy_user_dir = str(tmp_path / "legacy_user_tmp")
+    os.mkdir(legacy_user_dir)
+    return legacy_user_dir
 
-    def setUp(self):
-        super(TestSmartOSDataSource, self).setUp()
-
-        self.add_patch(DSMOS + ".get_smartos_environ", "get_smartos_environ")
-        self.add_patch(DSMOS + ".jmc_client_factory", "jmc_cfact")
-        self.legacy_user_d = self.tmp_path("legacy_user_tmp")
-        os.mkdir(self.legacy_user_d)
-        self.add_patch(
-            DSMOS + ".LEGACY_USER_D",
-            "m_legacy_user_d",
-            autospec=False,
-            new=self.legacy_user_d,
-        )
-        self.add_patch(
-            DSMOS + ".identify_file",
-            "m_identify_file",
-            return_value="text/plain",
-        )
 
-    def _get_ds(
-        self,
-        mockdata=None,
-        mode=DataSourceSmartOS.SMARTOS_ENV_KVM,
-        sys_cfg=None,
-        ds_cfg=None,
-    ):
-        self.jmc_cfact.return_value = PsuedoJoyentClient(mockdata)
-        self.get_smartos_environ.return_value = mode
+@pytest.fixture
+def m_jmc_client_factory(mocker):
+    return mocker.patch(
+        DSMOS + ".jmc_client_factory", return_value=PsuedoJoyentClient(None)
+    )
```

Comment:
This set of fixtures are a great example of the clarity, simplification and savings of fixtures through pytest. Thanks! 

Given that most tests which use this fixture overwrite the return_value and provide PsuedoJoyentClient)MOCK_RETURNS). Does it make more sense to provide MOCK_RETURNS here and only override in the test cases which need to provide a subset of `my_returns` ?

```suggestion
@pytest.fixture
def m_jmc_client_factory(mocker):
    return mocker.patch(
        DSMOS + ".jmc_client_factory", return_value=PsuedoJoyentClient(MOCK_RETURNS)
    )
```

Reply:
Much cleaner, there wasn't a case using the `None` value. Thanks.